### ISSUE TITLE
EM Module objects should be able to have tensor material coefficients

### ISSUE THREAD

**Issue by cticenhour** [idaholab/moose#21206]
## Reason
Cold magnetized plasma wave calculations utilize a tensor for the electric conductivity (Stix tensor), while applications like Ferret require d**P**/dt and d**M**/dt  (time derivatives of the polarization and magnetization vectors of a material) as part of Maxwell's Equations for their calculations. At the moment, the electromagnetics module mostly utilizes functions and basic AD material properties for quantities like this. They should be swapped out with (or given the ability to take in) tensors for these properties.

## Design
Relevant objects in the EM module should have the ability to use tensors for their material properties where necessary. 

## Impact
New capability

@mangerij please add additional notes for your use case in this ticket if you feel more explanation or notes relevant to implementation is necessary. Thanks!

**Comment by mangerij** [comment]
Yeah, just to add some notes and my naive understanding of this... 

Since there are many possible "modes" of the EM module, I think there will be some confusion on how to consistently have all these objects talking to each other.

In the example of d**P**/dt, this defines a dielectric tensor since _weak_ electric fields propagating parallel or perpendicular to the initial direction of **P** will induce a different induced harmonic response across the frequency band. In this "mode", one could extract the E(M)-induced dielectric constant across frequency. In this case, one doesn't need the dielectric tensors as input but they would be calculated from the module.

But since in the case of ferroelectrics **P** is defined as a continuum variable that has both ionic and electronic contributions, this frequency dispersion can have orders of magnitude different response depending on the strength or frequency of the electric field. For example, if the wave has significantly high (optical) frequency, then the dielectric tensor components can be considered as static numbers in a transient EM simulation and the ground state of **P** from d**P**/dt simulations then defines the refractive index that these high frequency waves will see.  This would be a different "mode" of calculations that also could be interesting more akin to wave-guide simulations.

I think for now, it will just be useful that these quantities (permittivity, conductivity, refractive index) are tensor objects that can be functions of other variables. It will be interesting to see if MOOSE can handle EM simulations where these (possibly frequency-dependent) objects vary drastically in space on the order of the wavelength of the EM wave.
